src/main/java/com/klodnicki/Bike/v2/service/BikeService.java

```diff
@@ -1,7 +1,103 @@
 package com.klodnicki.Bike.v2.service;
 
+import com.klodnicki.Bike.v2.DTO.bike.BikeForAdminResponseDTO;
+import com.klodnicki.Bike.v2.DTO.bike.BikeRequestDTO;
+import com.klodnicki.Bike.v2.model.entity.Bike;
+import com.klodnicki.Bike.v2.repository.BikeRepository;
+import org.modelmapper.ModelMapper;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
 @Service
-public class BikeService {
+//public class BikeService implements GenericService<BikeForAdminResponseDTO, Bike> {
+//zamienilismy BikeService dependency wstrzykiwane w controllerze na GenericBikeService
+//Żeby móc podstawić różne implementacje BikeService, nie tylko ten jeden konkretny BikeService. Jak zrobisz klasę
+//        BetterBikeService możesz bez żadnych zmian w kontrolerze z niego korzystać od razu
+public class BikeService implements GenericBikeService {
+
+    private final BikeRepository bikeRepository;
+
+    ModelMapper modelMapper = new ModelMapper();
+
+    public BikeService(BikeRepository bikeRepository) {
+        this.bikeRepository = bikeRepository;
+    }
+
+    @Override
+    public BikeForAdminResponseDTO add(BikeRequestDTO bikeRequestDTO) {
+        //using ModelMapper to convert(map) bikeRequestDTO into bike. I need Bike to be able to save it in repo
+        Bike bike = modelMapper.map(bikeRequestDTO, Bike.class);
+        Bike savedBike = bikeRepository.save(bike);
+        //converting back to BikeForAdminResponseDTO using modelMapper
+        BikeForAdminResponseDTO bikeDto = modelMapper.map(savedBike, BikeForAdminResponseDTO.class);
+
+        return bikeDto;
+    }
+
+    @Override
+    public BikeForAdminResponseDTO findById(Long id) {
+        Bike savedBike = bikeRepository.findById(id).orElseThrow(IllegalArgumentException::new); //TODO: add custom exception
+
+        return modelMapper.map(savedBike, BikeForAdminResponseDTO.class);
+    }
+
+    @Override
+    public List<BikeForAdminResponseDTO> findAll() {
+        List<BikeForAdminResponseDTO> bikeListDto = new ArrayList<>();
+        Iterable<Bike> bikeList = bikeRepository.findAll();
+
+        for (Bike bike : bikeList) {
+
+            // Builder is in Lombok dependency
+            // BikeForAdminResponseDTO must have @Builder annotation
+            BikeForAdminResponseDTO bikeDTO = BikeForAdminResponseDTO.builder()
+                    .id(bike.getId())
+                    .serialNumber(bike.getSerialNumber())
+                    .isRented(bike.isRented())
+                    .bikeType(bike.getBikeType())
+                    .rentalStartTime(bike.getRentalStartTime())
+                    .rentalEndTime(bike.getRentalEndTime())
+                    .amountToBePaid(bike.getAmountToBePaid())
+                    .gpsCoordinates(bike.getGpsCoordinates())
+                    .build();
```

Comment:
What would be even better and cleaner way than using builder for this case?

Reply:
modelMapper! 🗡️